File: frontend/src/types/item.ts
export interface ItemBase {
  title: string;
  description?: string;
  image_url?: string;
}

export type ItemCreate = ItemBase;

export interface ItemResponse extends ItemBase {
  id: number;
  owner_id: number;
}

export interface GetItemsParams {
  skip?: number;
  limit?: number;
  search?: string;
}

File: backend/app/schemas/item.py
from pydantic import BaseModel
from typing import Optional

class ItemBase(BaseModel):
    title: str
    description: Optional[str] = None
    image_url: Optional[str] = None

class ItemCreate(ItemBase):
    pass

class ItemResponse(ItemBase):
    id: int
    owner_id: int

    class Config:
        from_attributes = True
